.github/workflows/backport-workflow.yml

```diff
@@ -29,15 +37,36 @@ jobs:
         if: ${{ steps.check_pr_labels.outputs.result == 'true' }}
         uses: actions/checkout@v4
         with:
+          # ensure the backport target branch is checked out, too
           fetch-depth: 0
+          path: repo
+
+      - name: Checkout Backport tool
+        if: ${{ steps.check_pr_labels.outputs.result == 'true' }}
+        uses: actions/checkout@v4
+        with:
+          repository: hazelcast/backport
+          path: backport
```

Comment:
> You mean if https://github.com/hazelcast/backport was a composite action?
You'd still need to checkout the repo to run the script(s), but that could be baked into a GitHub action to abstract that from the caller.

Yes,  which is better as we don't need to worry about checkout (so less code lines) and feels much cleaner. So if `hazelcast/backport` provides a wrapper action then it will simplify things I think. Up to you though!!

> There's not too much duplication here, but if you think it's of value let me know and I can try again.

Yes I think lets try but lets not spend too much time. If we can consolidate that will be good for sure

Reply:
> > You mean if https://github.com/hazelcast/backport was a composite action?
> > You'd still need to checkout the repo to run the script(s), but that could be baked into a GitHub action to abstract that from the caller.
> 
> Yes, which is better as we don't need to worry about checkout (so less code lines) and feels much cleaner. So if `hazelcast/backport` provides a wrapper action then it will simplify things I think. Up to you though!!
> 
> > There's not too much duplication here, but if you think it's of value let me know and I can try again.
> 
> Yes I think lets try but lets not spend too much time. If we can consolidate that will be good for sure

I've tried, but I'm not sure it's an improvement. Opinions?
- https://github.com/JackPGreen/backport/commit/70ac1b036a6452cfb06f4be83693aae0ba3ccc08
- https://github.com/JackPGreen/hz-docs/commit/f994d73f694ec6b363ef6551cf3ba1c0fc9ca70b
- https://github.com/JackPGreen/hazelcast-mono/commit/599876613a2205fca2308bbbc0936ab07bd91033

Reply:
thanks. Personally I think its an improvement. 

1. Its apparent its reduced duplication which is a plus
2. No need to checkout
3. Less code overall
4. Using Actions is the standard way to share common code in GH and we can version in the future if need be
5. Any future issues will only need to be fixed in one place, although arguably, any changes made will need to ensure does not break the callers

So if this actually works then I think we should make the changes but I leave it to you!

ps: Thanks for trying this out btw.

Reply:
> I think its an improvement.
> 
> {...}
> 
> So if this actually works then I think we should make the changes but I leave it to you!

- 6623a3e26d2d5307afb55185ef8329ebc194b969
- https://github.com/hazelcast/backport/pull/17
- https://github.com/hazelcast/hazelcast-mono/pull/3642

Reply:
thanks